File: webflux-dialogue/src/testFixtures/java/com/study/webflux/rag/fixture/RagDialogueRequestFixture.java
package com.study.webflux.rag.fixture;

import java.time.Instant;

import com.study.webflux.rag.application.dialogue.dto.RagDialogueRequest;

public final class RagDialogueRequestFixture {

	public static final String DEFAULT_TEXT = "테스트 질문입니다";

	private RagDialogueRequestFixture() {
	}

	public static RagDialogueRequest create() {
		return new RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, DEFAULT_TEXT,
			Instant.now());
	}

	public static RagDialogueRequest createWithText(String text) {
		return new RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, text,
			Instant.now());
	}

	public static RagDialogueRequest createWithSessionId(String sessionId) {
		return new RagDialogueRequest(sessionId, DEFAULT_TEXT, Instant.now());
	}

	public static RagDialogueRequest createWithBlankText() {
		return new RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, "",
			Instant.now());
	}

	public static RagDialogueRequest createWithNullTimestamp() {
		return new RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, DEFAULT_TEXT,
			null);
	}
}

File: app-miyou/src/testFixtures/kotlin/com/miyou/app/fixture/RagDialogueRequestFixture.kt
package com.miyou.app.fixture

import com.miyou.app.infrastructure.inbound.web.dialogue.dto.RagDialogueRequest
import java.time.Instant

object RagDialogueRequestFixture {
    const val DEFAULT_TEXT = "안녕하세요"

    @JvmStatic
    fun create(): RagDialogueRequest =
        RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, DEFAULT_TEXT, Instant.now())

    @JvmStatic
    fun createWithText(text: String): RagDialogueRequest =
        RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, text, Instant.now())

    @JvmStatic
    fun createWithSessionId(sessionId: String): RagDialogueRequest =
        RagDialogueRequest(sessionId, DEFAULT_TEXT, Instant.now())

    @JvmStatic
    fun createWithBlankText(): RagDialogueRequest =
        RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, "", Instant.now())

    @JvmStatic
    fun createWithNullTimestamp(): RagDialogueRequest =
        RagDialogueRequest(ConversationSessionFixture.DEFAULT_SESSION_ID, DEFAULT_TEXT, Instant.now())
}
